### PR TITLE
TMDM-11709 When saving datamodel and importing datamodel from local, do check System role setting permissions.

### PR DESCRIPTION
**Pull request by sbliu**
jira: https://jira.talendforge.org/browse/TMDM-11709

**What is the current behavior?** (You should also link to an open issue here)
Do not check system role permissions set on data model .

**What is the new behavior?**
Do check system role permissions set on data model when saving data model and importing data model from local .

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by sbliu** [comment]
This is a new created PR, original one has been discarded and commits has been cherry pick to this.
Main comment from original PR:

pbabin commented 20 days ago
@sbliu Given your PR title, it doesn't check validity of a model imported from a MDM server. Is that the case?

sbliu commented 20 days ago •  edited 
@pbabin Yes, does not check the validity of model when importing from server, this is a feature, so a reasonable assumption is server side data model does not own validation error, if have validation errors, data model can not be deployed.

**Comment by hbhong** [comment]
LGTM

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/validation/PermissionValidationRule.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+import org.talend.mdm.commmon.util.core.PermissionConstants;
+import org.w3c.dom.Element;
+
+/**
+ * @author sbliu
+ *
+ */
+public class PermissionValidationRule implements ValidationRule {
+    
+    private final String ELEMENT_TYPE_ENTITY = "entity"; //$NON-NLS-1$
```
Comment (pbabin) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/validation/PermissionValidationRule.java]: static?

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/PermissionConstants.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    String PERMISSIONTYPE_HIDE = "No Access";
+    String PERMISSIONTYPE_DENY_CREATE = "No Create Access";
+    String PERMISSIONTYPE_DENY_DELETE_PHYSICAL = "No Physic Delete";
+    String PERMISSIONTYPE_DENY_DELETE_LOGICAL = "No Logical Delete";
+    String PERMISSIONTYPE_WORKFLOW_ACCESS = "Workflow Access";
+    //
+    String VALIDATION_PERMISSION_MARKER = "validation.permission.validated";
+    //
+    String PERMISSION_ROLE_SYSTEM_PREFIX = "system_";
+    String PERMISSION_ROLE_ADMINISTRATION = "administration";
```
Comment (pbabin) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/PermissionConstants.java]: Aren't these already defined in ICoreConstants?

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/validation/PermissionValidationRule.java (hunk 1 above)
Comment (sbliu) [reply main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/validation/PermissionValidationRule.java]: Not static

Comment (pbabin) [reply main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/validation/PermissionValidationRule.java]: Why isn't it a constant?

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/PermissionConstants.java (hunk 2 above)
Comment (sbliu) [reply main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/util/core/PermissionConstants.java]: Dup constants removed.

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/validation/PermissionValidationRule.java (hunk 1 above)
Comment (sbliu) [reply main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/validation/PermissionValidationRule.java]: No why, not a constant is also ok, as your hope, i can change it to constant.

**Pull request merged**